cns/restserver/internalapi.go

```diff
@@ -389,3 +389,61 @@ func (service *HTTPRestService) CreateOrUpdateNetworkContainerInternal(req *cns.
 
 	return returnCode
 }
+
+// GetHomeAz gets home AZ from cache or from nmagent
+func (service *HTTPRestService) GetHomeAz(ctx context.Context) cns.GetHomeAzResponse {
+	homeAz := service.readHomeAzCache()
+	if homeAz != "" {
+		return cns.GetHomeAzResponse{HomeAzResponse: nmagent.HomeAzResponse{HomeAz: homeAz}}
+	}
+	suppApisCache := service.readSupportedApisCache()
+	if len(suppApisCache) == 0 || !isAPISupportedByNMAgent(suppApisCache, getHomeAzAPIName) { // to account for if nma getting updated underneath
+		// Get NMAgent supported apis
+		supportedAPIs, err := service.nma.SupportedAPIs(ctx)
```

Comment:
can code between lines 401 - 410 be extracted into a separate func like updateNMASupportedAPICache? The goal of that func is to query NMA and update the cache.